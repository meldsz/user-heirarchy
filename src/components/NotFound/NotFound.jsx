import { useNavigate } from "react-router";

export const NotFound = () => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate("/");
  };

  return (
    <div data-testid="not-found-container">
      <h1> Oops! The page you're looking for doesn't exist.</h1>
      <button data-testid="home-btn" onClick={handleClick}>
        Back to Home
      </button>
    </div>
  );
};
