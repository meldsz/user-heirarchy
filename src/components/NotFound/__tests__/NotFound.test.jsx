import { testRender } from "../../../common/tests/helper";
import { NotFound } from "../NotFound";
import { fireEvent } from "@testing-library/dom";

describe("<NotFound/>", () => {
  it("should render correctly", () => {
    const { getByTestId, asFragment } = testRender(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("not-found-container")).toBeDefined();
  });
  it("should navigate to login page correctly", () => {
    const { getByTestId } = testRender(<NotFound />);
    fireEvent.click(getByTestId("home-btn"));
  });
});
