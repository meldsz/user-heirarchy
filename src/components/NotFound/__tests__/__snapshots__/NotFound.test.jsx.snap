// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<NotFound/> should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="not-found-container"
  >
    <h1>
       Oops! The page you're looking for doesn't exist.
    </h1>
    <button
      data-testid="home-btn"
    >
      Back to Home
    </button>
  </div>
</DocumentFragment>
`;
